src/main/java/se/liu/ida/hefquin/engine/queryplan/utils/TextBasedLogicalPlanPrinterImpl2.java

```diff
@@ -48,11 +71,43 @@ public String getIndentLevelString(final int planNumber, final int planLevel, fi
 	 */
 	public void planWalk( final LogicalPlan plan, final int planNumber, final int planLevel, final int numberOfSiblings, final PrintStream out) {
 		final String indentLevelString = getIndentLevelString(planNumber, planLevel, numberOfSiblings);
-		out.append( indentLevelString + plan.getRootOperator().toString() );
-		out.append( System.lineSeparator() );
+		final LogicalOperator rootOp = plan.getRootOperator();
+		if ( rootOp instanceof LogicalOpMultiwayJoin ) {
+			printOperatorInfoMultiwayJoin( (LogicalOpMultiwayJoin) rootOp, out, indentLevelString );
+		}
+		else if (rootOp instanceof LogicalOpMultiwayUnion) {
+			printOperatorInfoMultiwayUnion( (LogicalOpMultiwayUnion) rootOp, out, indentLevelString );
+		}
+		else if (rootOp instanceof LogicalOpRequest) {
+			String indentLevelStringForOpDetail = getIndentLevelStringForDetail(planNumber, planLevel, numberOfSiblings);
```

Comment:
I guess that you can move this outside of this particular else-branch, before the sequence of if-else statements, because you may need it for other types of operators as well.

Additionally, the variable should be marked as `final`.